src/shogun/kernel/GaussianARDKernel.cpp

```diff
@@ -65,22 +64,9 @@ CGaussianARDKernel* CGaussianARDKernel::obtain_from_generic(CKernel* kernel)
 
 float64_t CGaussianARDKernel::compute(int32_t idx_a, int32_t idx_b)
 {
-	REQUIRE(lhs && rhs, "Features not set!\n")
```

Comment:
the require is not necessray because this is a private method?
Or do you simply check at the places where the user defines the features?